### PR TITLE
Weather app

### PR DESCRIPTION
**Pull request by okikiola11**
I worked on the following:
- Implemented async and await functionality
- Implemented the search functionality for a given city location by a user
- Added the functionality to switch from Celsius to Fahrenheit in degrees

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "js-weather-app",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "dev": "webpack --mode development",
+    "build": "webpack",
```
Comment (addod19) [inline package.json]: Let us kindly fix our `build` script as we did to the `dev` script. Kindly add `--mode production` to the build script as we see in the dev script :muscle:

File: package.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "eslint": "^6.8.0",
+    "eslint-config-airbnb-base": "^14.1.0",
+    "eslint-plugin-import": "^2.20.2",
+    "style-loader": "^2.0.0",
+    "stylelint": "^13.3.3",
+    "stylelint-config-standard": "^20.0.0",
+    "stylelint-csstree-validator": "^1.9.0",
+    "stylelint-scss": "^3.17.2",
+    "webpack": "^5.24.2",
+    "webpack-cli": "^4.5.0"
```
Comment (addod19) [inline package.json]: Kindly install webpack-dev-server, so we can have it as part of the devDependencies :smile:

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  showCity, showTempearture, showIcon,
+  showDescription, showHumidity, showWind, searchCity,
+  searchCityBtn, showCountry, weatherContainer,
+  showBodyBckImg, showTempDegButton,
+} from './dom';
+
+const API_KEY = '70c8519b5e37ccac35bcbea2d8b91b5b';
+let temperatureDegButton = showTempDegButton;
+
+async function fetchWeather(city) {
```
Comment (addod19) [inline src/index.js]: Kindly use ES6 arrow syntax instead of the `function` keyword we have currently :+1:

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    temp = (val * (9 / 5) + 32).toFixed(1);
+    type = '℉';
+  } else {
+    temp = ((val - 32) * (5 / 9)).toFixed(1);
+    type = '℃';
+  }
+  return { temp, type };
+};
+
+async function displayWeather(data) {
```
Comment (addod19) [inline src/index.js]: Kindly use ES6 arrow syntax instead of the `function` keyword we have currently :+1:

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    e.preventDefault();
+
+    const tempSwitch = temperatureSwitch(tempDataVal, temperatureDegButton.innerHTML);
+    tempDataVal = tempSwitch.temp;
+    temperatureDegButton.innerHTML = tempSwitch.type;
+    showTempearture.innerHTML = `${tempSwitch.temp}${tempSwitch.type}`;
+  });
+}
+
+async function search() {
```
Comment (addod19) [inline src/index.js]: Kindly use ES6 arrow syntax instead of the `function` keyword we have currently :+1:

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @okikiola11, congratulations on this project :fireworks: The application works as expected. The following requirements have been met:
- User can search for a specific location :100: 
- It has a toggle button to change between Celcius and Fahrenheit :champagne: 
- It changes appearance based on the received data :1st_place_medal: 
- Uses async and await :+1: 
Awesome :smile: 

However, there are a few changes we need to make before proceeding, kindly check the comments below:
- Kindly use ES6 arrow syntax instead of the `function` keyword
- Kindly add `webpack-dev-server` to the package.json file as devDependencies

Once these are resolved, kindly submit another code review request :slightly_smiling_face: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Review by mahmoud717 — CHANGES_REQUESTED** [review]
### Changes required ♻️ 

**Hello @okikiola11**

Great job with the project so far it looks amazing, but there are still some changes to be made:
 
---
### Requested 🔴 

- [ ] when you display the temperature, the decimal is moved one integer to the right, meaning that instead of 24.567 it's 245.67. please make sure to fix that.



---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

**Comment by okikiola11** [comment]
@mahmoud717 , please can you explain better, I can't see the error, I also can't find your name on slack

**Comment by mahmoud717** [comment]
This should be 28 instead of 281

**Comment by okikiola11** [comment]
Hi @mahmoud717, can you give me a little more explanation on what to do, because of the ${data. temp} value that I am getting from the API is what I am converting, which is giving me that result. How do I reduce the decimal value??

**Comment by mahmoud717** [comment]
Just divide it by ten.
281/10 = 28.1

**Comment by okikiola11** [comment]
@mahmoud717 , thank you. It's been resolved

**Review by ioilmio — CHANGES_REQUESTED** [review]
# Hello @okikiola11 👋🏽 
## You implemented all the changes required from previous TSEs, your project is looking, and the code is tidy and easy to understand congrats 🎉 

### There's just one big small problem:
### Your API key is publicly exposed, you have to remove it from your committed code.

### Here's is a [link](https://gist.github.com/derzorngottes/3b57edc1f996dddcab25) to guide you through the process, please read it and follow all the steps needed

*For any doubts please contact me on slack @illuminatosalvatore*

**Review by christianotieno — APPROVED** [review]
### Project Approved 🍾 
Hello @okikiola11 👋🏾 
##
This project has now met all the requirements needed for approval. :clap:
Congratulations on the hard work done with the changes and overall, on this milestone. Good job 👍🏾 
I will go ahead and mark this as complete. Good luck with the next tasks 🥇 
##
### Happy Coding 💻 
#### Christian Otieno (CR)
**| [Twitter](https://twitter.com/iamchrisotieno) | [Github](https://github.com/christianotieno) | [LinkedIn](https://www.linkedin.com/in/christianotieno/) |**

**Pull request merged**